assignment1/assignment.rb

```diff
@@ -0,0 +1,192 @@
+
+    palindrome == sentence_without_space ? 'true' : 'false'
+end
+
+palindrome "Never odd or even"
+
+# 9 is a Kaprekar number since 9 ^ 2 = 81 and 8 + 1 = 9, 297 is also Kaprekar number since 297 ^ 2 = 88209 and 88 + 209 = 297. In short, for a Kaprekar number k with n-digits, if you square it and add the right n digits to the left n or n-1 digits, the resultant sum is k. Find if a given number is a Kaprekar number.
+
+def kaprekar(k)
+    ks = k**2
+    lenf = (ks.to_s.length)-1
+    lenh = (ks.to_s.length / 2) - 1
+    a = ks.to_s[0..lenh].to_i
+    b = ks.to_s[lenh+1..lenf].to_i
+    k==(a+b) ? 'Kaprekar': 'Not Kaprekar'
+end
+
+kaprekar 297
+
+# Print prime numbers from the given array
+def prime_numbers num_arr
+    prime_arr = []
+    num_arr.each do |num|
+        count = 0
+        2.upto(num/2) do |i|
+            if (num%i==0)
+                count+=1
+            end
+        end
+        if (count.zero?)
+            prime_arr << num
+        end
+    end
+    prime_arr
+end
+
+prime_numbers [1, 2, 3, 4, 5, 7, 6, 9, 11, 12, 13]
+
+# Ruby program to find the prime numbers 
+# from the array
+
+arr = [12,45,23,39,37];
+
+i = 0;
+j = 0;
+flag = 0;
+        
+print "Prime Numbers are:\n";
+while(i<arr.size)
+    flag = 0;
+    j=2;
+    
+    while(j<arr[i]/2)
+        if(arr[i]%j==0)
+            flag=1;
+        end
+        j=j+1
+    end
+
+    if(flag == 0)
+        p arr[i]," ";
+    end
+    i=i+1		
+end
+
+
+# Write a program for fuel indication. Ex: 0 -> Out of fuel, 1-10 -> Low, Please fill, 11-30 -> Good for now, 31-50 -> Almost Full, > 50 -> Full
+
+def fuel_indication n
+    if (n == 0)
+        return 'Out of fuel'
+    elsif (1..10).include? n
+        return 'Low, Please fill'
+    elsif (11..30).include? n
+        return 'Good for now'
+    elsif (30..50).include? n
+        return 'Almost Full'
+    elsif (n > 50)
+        return 'Full'
+    end
+end
+
+fuel_indication 11
+
+# Swap the given values(without the third variable)
+
+a = 10
+b = 20
+
+a = a * b
+b = a/b
+a = a/b
+
+# Program to Randomly Select an Element From the Array
+
+arr = [1, 5, 6, 8,0]
+
+arr[rand(arr.length)]
+
+# Program to Count the Number of Each Vowel
+
+def vowel_count rand_str
+    vow_cnt = {};
+    vowels = ['a', 'e', 'i', 'o', 'u']
+    rnd_str_count = rand_str.length
+
+    vowels.each { |i|
+        vow_cnt[i] = (0..rand_str.length).find_all { |j| rand_str[j,1] == i}.count
+    }
+
+    vow_cnt
+end
+
+vowel_count "My name is Kiran"
+
+# Program to Sort a Hash by Value
+
+def hash_sort_value h
+    inv_hash = h.invert
+    inv_arr_keys = inv_hash.keys
+    sort_hash = {}
+
+    inv_arr_keys.each do |j, v|
+        0.upto(inv_arr_keys.size-2) do |i|
+            if inv_arr_keys[i] < inv_arr_keys[i+1]
+                inv_arr_keys[i], inv_arr_keys[i+1] = inv_arr_keys[i+1], inv_arr_keys[i]
+            end
+        end
+    end
+
+    inv_arr_keys.each do |i|
+        sort_hash[i] = inv_hash[i]
+    end
+
+    sort_hash.invert
+end
+
```

Comment:
This is sorting the hash correctly but in descending order, please check
I have checked this way 
=> :hash_sort_value
[198] pry(main)> h = {}
=> {}
[199] pry(main)> h[:a] = 1
=> 1
[200] pry(main)> h[:b]=98
=> 98
[201] pry(main)> h[:k]=2
=> 2
[202] pry(main)> h
=> {:a=>1, :b=>98, :k=>2}
[203] pry(main)> hash_sort_value h
=> {:b=>98, :k=>2, :a=>1}